common/service/ipmi/oem_1s_handler.c

```diff
@@ -283,6 +283,49 @@ __weak void OEM_1S_FW_UPDATE(ipmi_msg *msg)
 	return;
 }
 
+__weak void OEM_1S_GET_BIC_FW_INFO(ipmi_msg *msg)
+{
+	CHECK_NULL_ARG(msg);
+
+	if (msg->data_len != 1) {
+		msg->completion_code = CC_INVALID_LENGTH;
+		return;
+	}
+
+	msg->data_len = 0;
+
+	uint8_t component;
+	component = msg->data[0];
+	switch (component) {
+	case BIC_PLAT_NAME:
+		msg->data_len = strlen(PLATFORM_NAME);
+		memcpy(&msg->data[0], PLATFORM_NAME, msg->data_len);
+		break;
+
+	case BIC_PLAT_BOARD_ID:
+		msg->data_len = 1;
+		msg->data[0] = FIRMWARE_REVISION_1 & 0x0F;
+		break;
+
+	case BIC_PROJ_NAME:
+		msg->data_len = strlen(PROJECT_NAME);
+		memcpy(&msg->data[0], PROJECT_NAME, msg->data_len);
+		break;
+
+	case BIC_PROJ_STAGE:
+		msg->data_len = 1;
+		msg->data[0] = FIRMWARE_REVISION_1 & 0xF0;
+		break;
```

Comment:
Can we separate out FIRMWARE_REVISION_1 into two separate fields within plat_version.h and add a new enum for project stage

In a new file common/include/version.h
```
typedef enum {
        POC = 0,
        EVT,
        DVT,
        PVT,
        MP,
} PROJECT_STAGE;
```

Then in the plat_version file something like 
```
#define PROJECT_STAGE POC
#define BOARD_ID 0x01 // mainboard
...
...
#define FIRMWARE_REVISION_1 ((BOARD_ID & 0x0F) | (PROJECT_STAGE << 4))
```

Then we can use the PROJECT_STAGE and BOARD_ID defines where relevant and it's more consistent between platforms. (I can submit a PR after this one changing the other platforms to use a matching definition.)

Reply:
Sounds great to me, thanks a lot.
By the way, this new oem command is for sign-update use, so the platform name/board id/stage_id should follow sign tool **sign_bic_image.py** if any project need this feature.

Anyway this is just a notice for owner of project, not a common feature.